src/test/java/system_tests/TrialSeleniumTest.java

```diff
@@ -1,73 +1,208 @@
-
-    // create a test encoder
-    HashMap<String, String> encoderParam = new HashMap<>();
-    encoderParam.put("serialNumber", deviceEntity.getSerialNumber());
-    encoderParam.put("lastCommunication", "2020-11-28 09:40:00");
-
-    postRequest("http://localhost:8080/device", deviceParam);
-    postRequest("http://localhost:8080/encoder", encoderParam);
+  @AfterAll
+  static void tearDown() {
+    driver.quit();
   }
 
   @Test
-  void testAddEncoder() {
-    driver.get("http://localhost:3000/Devices");
-    driver.manage().window().setSize(new Dimension(782, 818));
+  @Order(1)
+  void testAddEncoder() throws IOException {
+    // Mock sender self-registration
+    postRequest("http://localhost:8080/device", testSenderDeviceParams);
+    postRequest("http://localhost:8080/encoder", testSenderEncoderParams);
 
-    driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+    // Go to List of Senders
+    driver.get("http://localhost:3000/Devices");
 
+    // Check for encoder
     WebElement encodersTable = driver.findElement(By.tagName("table")); // find encoders table
     List<WebElement> devicesRows =
         encodersTable.findElements(By.tagName("tr")); // find all tr elements inside found table
+    boolean assertValue =
+        devicesRows.stream()
+            .anyMatch(
+                row ->
+                    row.getText()
+                        .contains(EncoderFixture.getEncoderEntity1().getDevice().getDisplayName()));
+
+    assertTrue(assertValue);
+  }
+
+  @Test
+  @Order(2)
+  void testAddDecoder() throws IOException {
+    // mock receiver self-registration
+    postRequest("http://localhost:8080/device", testReceiverDeviceParams);
+    postRequest("http://localhost:8080/decoder", testReceiverDecoderParams);
+
+    // Go to List of Receivers
+    driver.get("http://localhost:3000/Devices");
+    driver.findElement(By.xpath("//button[@id='vertical-tab-1']/span")).click();
 
+    // Check for decoder
+    WebElement decodersTable = driver.findElement(By.tagName("table")); // find encoders table
+    List<WebElement> devicesRows =
+        decodersTable.findElements(By.tagName("tr")); // find all tr elements inside found table
     boolean assertValue =
-        devicesRows.stream().anyMatch(row -> row.getText().contains(deviceEntity.getDisplayName()));
+        devicesRows.stream()
+            .anyMatch(
+                row ->
+                    row.getText()
+                        .contains(DecoderFixture.getDecoderEntity2().getDevice().getDisplayName()));
 
     assertTrue(assertValue);
   }
 
-  @AfterEach
-  void tearDown() {
-    driver.quit();
+  @Test
+  @Order(3)
+  void testCreateStream() {
+    driver.get("http://localhost:3000/Streaming");
+    {
+      WebDriverWait wait = new WebDriverWait(driver, 100);
+      wait.until(
+          ExpectedConditions.elementToBeClickable(
+              By.cssSelector("#SenderTable .MuiList-root .MuiSvgIcon-root")));
+    }
```

Comment:
Any reason why we're using code blocks here? If it's just for readability I think that would be a good opportunity to add a helper method with a name.

Reply:
This done to have the variable local to the code block it’s defined within